src/client/resources/web/js/chat.mjs

```diff
@@ -122,33 +166,76 @@ function connect() {
         /** @type {string} */
         const rawJson = event.data;
         console.log(rawJson);
-		try {
-			const message = parseModServerMessage(rawJson);
-			// For now we only handle chat messages
-			if (message.type === 'chatMessage') {
-				displayChatMessage(message.payload, message.timestamp);
-			}
-		} catch (e) {
-			console.error('Error processing message:', e);
-		}
+        try {
+            const message = parseModServerMessage(rawJson);
+            // For now we only handle chat messages
+            if (message.type === 'chatMessage') {
+                // Skip if we've already seen this message
+                if (displayedMessageIds.has(message.payload.uuid)) {
+                    return;
+                }
+                displayedMessageIds.add(message.payload.uuid);
+                displayChatMessage(message.payload.component, message.timestamp, message.payload.history);
+            }
+            if (message.type === 'serverConnectionState') {
+                switch (message.payload) {
+                    case 'init':
+                        console.log('Received init event. It is something, init?');
```

Comment:
The init event might not be the right place to do this. I got it randomly while being connected to the server which cleared my history.